### PR TITLE
Components: Consolidate and add docs to Storybook

### PR DESCRIPTION
**Pull request by chad1008**
## What?
Consolidate multiple instances of key docs (`README.md` and `CONTRIBUTING.md`) and configure Storybook to display them.

## Why?
Up until now we've had two copies of both `README` and `CONTRIBUTING`: one in the package root, and another in the `src` directory. Consolidating these makes things clearer for contributors and surfacing them in Storybook makes them more visible and easier to access. Also because it was fun.

## How?
The two copies of `CONTRIBUTING` were identical, so the duplicate was just deleted.

The contents of the two `README` files were different, with the root copy focusing on detailed instructions for different aspects of a contribution, while the `src` copy focused more on the workflow of introducing a new component. I've combined all of this into a single file with one small change. The `src` copy's workflow for new components included a step to create a [Figma component](https://www.figma.com/file/ZtN5xslEVYgzU7Dd5CxgGZwq/WordPress-Components), which I don't believe we're still doing, so I took that step out. It's easy enough to add back in if it's still relevan.

Finally to render both of these docs in Storybook, I enabled the `transcludeMarkdown` config setting, which allows mdx files to import `.md` files and render their contents as a component. I then added two new `.mdx` files, one to import each of the desired docs.

## Testing Instructions
Look over both affected docs to make sure the relevant info has been carried over. Check that the both render nicely in Storybook.

### CODE REVIEW

File: storybook/stories/docs/contributing.story.mdx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { Meta } from '@storybook/addon-docs';
+import Contributing from '../../../packages/components/CONTRIBUTING.md';
+
+<Meta title="Docs/Contributing Guidelines" />
+
+<Contributing />
```
Comment (mirka) [inline storybook/stories/docs/contributing.story.mdx]: Cool 💛

**Review by mirka — APPROVED** [review]
Nice, love this!

While reviewing I had some minor suggestions that I couldn't make directly on this PR, so I [branched off and made them](https://github.com/WordPress/gutenberg/compare/components-consolidate-docs...components-consolidate-docs-tweaks?expand=1), if that's ok. The main thing I'd like to suggest is to move the docs to the Components section in the Storybook, because our Storybook is starting to get pretty multi-tenant (block-editor, edit-site, etc).

Components docs live under the Components section in the Storybook

What do you think? Feel free to tweak or cherry pick from my branch as you see fit.

**Comment by chad1008** [comment]
That all makes sense, thanks for sharing the changes @mirka! I noticed that when moving the README into the `Components` section, you set the doc title to `Introduction`, so we'd have two of those, one in `Docs` and one in `Components`. Was that intentional?

**Comment by mirka** [comment]
> I noticed that when moving the README into the `Components` section, you set the doc title to `Introduction`, so we'd have two of those, one in `Docs` and one in `Components`. Was that intentional?

Yes, I guess because "readme" as a naming convention is more for when you're browsing files, not when browsing online documentation. No strong opinion though!

**Comment by chad1008** [comment]
Thanks again @mirka. Cherry-picked and merging ✅

**Pull request merged**